archspec/cpu/microarchitecture.py

```diff
@@ -218,15 +229,9 @@ def satisfies_constraint(entry, version):
             min_version, max_version = entry["versions"].split(":")
 
             # Check version suffixes
-            min_version, min_suffix = version_components(min_version)
-            max_version, max_suffix = version_components(max_version)
-            version, suffix = version_components(version)
-
-            # If the suffixes are not all equal there's no match
-            if (suffix != min_suffix and min_version) or (
-                suffix != max_suffix and max_version  # pylint: disable=bad-continuation
-            ):
-                return False
+            min_version, _ = version_components(min_version)
+            max_version, _ = version_components(max_version)
+            version, _ = version_components(version)
```

Comment:
I'm a little confused by this. Why use `version_components()` at all now that we do not care about suffixes? Why ignore the suffix and not include it in the tuple like we do in Spack?

Reply:
> Why use version_components() at all now that we do not care about suffixes?

To minimize code changes with respect to the previous version. I can change this to not use `version_components` at all if it seems better. 

> Why ignore the suffix and not include it in the tuple like we do in Spack?

Before this PR a suffix would have identified a completely different "branch" of a compiler. This means that `clang@X.Y.Z` and `clang@X.Y.Z-apple` would be looking for different entries in the JSON file to check which optimization flags should be used. This is because  before a check on the version range we check that the suffixes match. 

This convention however is very much Spack specific and we are employing it in `archspec` only because of Apple's Clang. We also had a couple of reports from users in Spack saying that this logic is preventing them from tagging manually with different dash separated suffixes a version of e.g. `gcc` as a mnemonic name for compilers that injects custom flags. For instance `gcc@9.2.0-optimized` wouldn't inject the correct microarchitecture specific optimization flags since we have no information in the JSON file on the `-optimized` suffix of GCC.

Since Apple's Clang is now a separate compiler in the most recent version of the JSON file, we have no reason to rely in `archspec` on custom suffixes and to the best of my knowledge no compiler we encountered uses version suffixes to distinguish among different flavors. Thus this changeset disregards suffixes entirely and relies only on version numbers to determine the correct optimization flags.

Reply:
Ok works for me. We may want to consider special suffixes better in the future (like a b and rc), but we don’t actually do that in spack right now either.